File: LeetCode/LeetCode Practice Problems/23- Merge k Sorted Lists.cpp
class Solution {
public:
    ListNode* mergeKLists(vector<ListNode*>& lists) {
        vector<int> a;
        for(int i=0;i<lists.size();i++)
        {
            ListNode* t = lists[i];
            while(t!=NULL)
            {
                a.push_back(t->val);
                t = t->next;
            }
        }
        
        sort(a.begin() , a.end());
        ListNode* head = new ListNode(0);
        ListNode* t = head;
        for(auto i : a)
        {
            t->next = new ListNode(i);
            t = t->next;
        }
        
        return head->next;
    }
};

File: LeetCode/LeetCode Practice Problems/23-Merge k Sorted Lists.java
class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        ListNode list = new ListNode(-1);
        ListNode tmp = list;
        int idx = -1;
        
        do {
            idx = -1;
            int val = Integer.MAX_VALUE;
            for(int i=0;i<lists.length;i++) {
                if(lists[i] == null) continue;
                if(val > lists[i].val) {
                    idx = i;
                    val = lists[i].val;
                }
            }
            
            if(idx != -1) {
                tmp.next = new ListNode(lists[idx].val);
                tmp = tmp.next;
                lists[idx] = lists[idx].next;
            }
        } while(idx != -1);
        
        return list.next;
    }
}
